ui/lib/client/index.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react'
+import i18next from 'i18next'
+import axios from 'axios'
+import { message, Modal, notification } from 'antd'
+import * as singleSpa from 'single-spa'
+
+import * as auth from '@lib/utils/auth'
+import * as routing from '@lib/utils/routing'
+import * as i18n from '@lib/utils/i18n'
+import publicPathPrefix from '@lib/utils/publicPathPrefix'
+
+import { DefaultApi } from './api'
+
+export * from './api'
+
+//////////////////////////////
+
+let basePath: string
+let apiClientInstance: DefaultApi
+
+function save(instanceBasePath: string, instance: DefaultApi) {
+  basePath = instanceBasePath
+  apiClientInstance = instance
+}
+
+function getInstance(): DefaultApi {
+  return apiClientInstance
+}
+
+function getBasePath(): string {
+  return basePath
+}
+
+export default { getInstance, getBasePath }
+
+//////////////////////////////
+
+export enum ErrorStrategy {
+  Default = 'default',
+  Custom = 'custom',
+}
+
+function initAxios() {
+  i18n.addTranslations(require.context('./translations/', false, /\.yaml$/))
+
+  const instance = axios.create()
+  instance.interceptors.response.use(undefined, function (err) {
+    const { response, config } = err
+    const errorStrategy = config.errorStrategy as ErrorStrategy
+    const method = (config.method as string).toLowerCase()
+
+    let errCode: string
+    if (err.message === 'Network Error') {
+      errCode = 'error.network'
+    } else {
+      errCode = response?.data?.code || err.message || 'error.api.other'
+    }
+    const content = i18next.t(errCode)
+    err.msg = content
+
+    if (errCode === 'error.api.unauthorized') {
+      // Handle unauthorized error in a unified way
+      if (!routing.isLocationMatch('/') && !routing.isSignInPage()) {
+        message.error({ content, key: errCode })
+      }
+      auth.clearAuthToken()
+      singleSpa.navigateToUrl('#' + routing.signInRoute)
+      err.handled = true
+    } else if (errorStrategy === ErrorStrategy.Default) {
+      if (method === 'get') {
```

Comment:
In what case will we fall into this `if` branch now?

Reply:
This is an example from InstancesProfiling/pages/Detail.tsx:

```
  const handleRowClick = usePersistFn(
    async (rec, _idx, _ev: React.MouseEvent<HTMLElement>) => {
      const res = await client
        .getInstance()
        .getActionToken(rec.id, 'single_view')
      const token = res.data
      if (!token) {
        return
      }
      window.open(
        `${client.getBasePath()}/profiling/single/view?token=${token}`,
        '_blank'
      )
    }
  )
```

This is a GET request and we don't use `useClientRequest()`.